test/e2e/functional/tests/e2e.go

```diff
@@ -1126,29 +1137,20 @@ var _ = Describe("metallb", func() {
 			err = testclient.Client.ConfigMaps(OperatorNameSpace).Delete(context.Background(), consts.MetalLBConfigMapName, metav1.DeleteOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() (string, error) {
+			Eventually(func() bool {
 				var err error
-				configMap, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
-- name: addresspool1
-  protocol: layer2
-  addresses:
-  - 1.1.1.1-1.1.1.100
-`))
+				_, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+				return k8serrors.IsNotFound(err)
+			}, metallbutils.Timeout, metallbutils.Interval).Should(Equal(true))
 
 			By("Checking AddressPool resource is deleted and ConfigMap is cleared")
 			err = testclient.Client.Delete(context.Background(), addressPool)
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() string {
-				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return configmap.Data[consts.MetalLBConfigMapName]
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML("{}"))
+			Eventually(func() bool {
+				_, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+				return k8serrors.IsNotFound(err)
```

Comment:
Why this? Shouldn't the configmap be cleared?

Reply:
it was a regression due to previous change, fixed that and restored to same test now.